### PR TITLE
Feature 23/select date to submit document

### PR DESCRIPTION
**Pull request by aufgfua**
## Description

Create page to select referral sheet send date

## Changes

Create the referral sheet send date page
Change AppCalendar structure to better match the design (icon on the right and 0.5 opacity on invalid dates)
Change AppDialog to enable having two buttons

## Checks

- [x] Project Builds
- [x] Project passes tests and checks
- [ ] Updated documentation accordingly

## Additional information

[Issue #23](https://github.com/TheTributeCommunity/fesbal-frontend/issues/23)

Just left a TODO to mark the place where we should send the date to the server

image
image
image

### CODE REVIEW

File: src/pages/RegisterReferralSendDate.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import AppReferralSendDateForm from "../components/molecules/AppReferralSendDateForm";
+import AppPageHeader from "../components/molecules/AppPageHeader";
+import AppWrapper from "../components/molecules/AppWrapper";
+import {AppRoute} from "../enums/app-route";
+import AppMessageDialog from "../components/molecules/AppMessageDialog";
+import AlertIcon from "../components/icons/AlertIcon";
+
+const RegisterReferralSendDate = () => {
+    const {t: translate} = useTranslation(namespaces.pages.registerReferralSendDate);
+    const [showConfirmDialog, setshowConfirmDialog] = useState(false);
```
Comment (Saixel) [inline src/pages/RegisterReferralSendDate.tsx:13]:
```suggestion
    const [showConfirmDialog, setShowConfirmDialog] = useState(false);
```

File: src/i18n/pages/registerReferralSendDate.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+export const registerReferralSendDateEs = {
+    headerTitle: "Hoja de derivación",
+    title: "Cita con los Servicios Sociales",
+    description: `Por favor indica la fecha de la cita con Servicios Sociales. Si aún no la tiene por favor pongáse en contacto con los Servicios Sociales de su comunidad.`,
+    sendDate: "Fecha de la cita",
+    
+    adviceTitle: "Importante",
+    adviceDescription: "Recuerda que una vez cumplido la fecha tendrás que subir la hoja de derivación que te entreguen. De lo contrario se bloqueará el servicio.",
```
Comment (Saixel) [inline src/i18n/pages/registerReferralSendDate.ts]:
```suggestion
    adviceDescription: "Recuerda que una vez cumplida la fecha tendrás que subir la hoja de derivación que te entreguen. De lo contrario se bloqueará el servicio.",
```

File: src/pages/RegisterReferralSendDate.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    const onSubmit = () => {
+        setshowConfirmDialog(true);
```
Comment (Saixel) [inline src/pages/RegisterReferralSendDate.tsx:16]:
```suggestion
        setShowConfirmDialog(true);
```

File: src/pages/RegisterReferralSendDate.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    };
+
+    const selectSubmit = () => {
+        // TODO: submit form
+        setshowConfirmDialog(false);
```
Comment (Saixel) [inline src/pages/RegisterReferralSendDate.tsx:21]:
```suggestion
        setShowConfirmDialog(false);
```

File: src/pages/RegisterReferralSendDate.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    };
+
+    const selectCancel = () => {
+        setshowConfirmDialog(false);
```
Comment (Saixel) [inline src/pages/RegisterReferralSendDate.tsx:25]:
```suggestion
        setShowConfirmDialog(false);
```

File: src/components/molecules/AppMessageDialog.tsx [hunk 6]
```diff
@@ -3,44 +3,63 @@ import AppNextButton from "../atom/AppNextButton";
...
 }) => {
     return (
-        <div className="absolute top-0 left-0 bg-secondary-color bg-opacity-40 z-50 w-full h-full">
-            <div className="absolute bottom-6 left-0 right-0 px-4 ">
-                <div className="w-full md:w-1/2 lg:w-1/3 p-6 pt-10 text-center text-secondary-color font-roboto-flex bg-white rounded-2xl mx-auto">
-                    <div className="flex flex-col">
-                        <div className="mb-6 mx-auto">
-                            {icon}
+        <>
+            { visible &&
```
Comment (Saixel) [inline src/components/molecules/AppMessageDialog.tsx:35]:
```suggestion
            {visible &&
```

File: src/components/molecules/AppMessageDialog.tsx [hunk 7]
```diff
@@ -3,44 +3,63 @@ import AppNextButton from "../atom/AppNextButton";
...
+                                <div className="mb-6 mx-auto">
+                                    {icon}
+                                </div>
+                                <h2 className={titleClassname + " mb-8 font-big-title"}>{title}</h2>
+                            
+                                <p className={descriptionClassname + " font-light mb-12 text-base"}>{description}</p>
+                            
+                            </div>
+                            <div className="flex gap-4">
+                                { secondaryButtonText &&
```
Comment (Saixel) [inline src/components/molecules/AppMessageDialog.tsx:49]:
```suggestion
                                {secondaryButtonText &&
```

**Review by Saixel — COMMENTED** [review]
Just some suggestions for formatting (and my OCD) :joy:

File: src/pages/RegisterReferralSendDate.tsx (hunk 1 above)
Comment (aufgfua) [reply src/pages/RegisterReferralSendDate.tsx:13]: Nice catch again! I definitely didn't see that

**Review by Saixel — APPROVED** [review]
It would be good to review the formatting/identation, but approved.

**Pull request merged**